src/Humanizer/Localisation/NumberToWords/FrenchAbstractNumberToWordsConverter.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Humanizer.Localisation.NumberToWords
+{
+    internal abstract class FrenchAbstractNumberToWordsConverter : GenderedNumberToWordsConverter
+    {
+        protected static readonly string[] UnitsMap = new string[] { "zéro", "un", "deux", "trois", "quatre", "cinq", "six", "sept", "huit", "neuf", "dix", "onze", "douze", "treize", "quatorze", "quinze", "seize", "dix-sept", "dix-huit", "dix-neuf" };
+
+        protected FrenchAbstractNumberToWordsConverter()
+        {
+
+        }
+
+        public override string Convert(int number, GrammaticalGender gender)
+        {
+            if (number == 0)
+                return UnitsMap[0];
+
+            if (number < 0)
+                return string.Format("moins {0}", Convert(Math.Abs(number)));
+
+            var parts = new List<string>();
+
+            if ((number/1000000000) > 0)
+            {
+                parts.Add(string.Format("{0} milliard{1}",
+                    Convert(number/1000000000),
+                    number/1000000000 == 1 ? string.Empty : "s"));
+
+                number %= 1000000000;
+            }
+
+            if ((number/1000000) > 0)
+            {
+                parts.Add(string.Format("{0} million{1}",
+                   Convert(number/1000000),
+                   number/1000000 == 1 ? string.Empty : "s"));
+
+                number %= 1000000;
+            }
+
+            if ((number/1000) > 0)
+            {
+                parts.Add(string.Format("{0}mille",
```

Comment:
@hazzik I dont know. It should be, maybe. I refactored. I didn't change the original logic. Tests should check this ;)